hosted/eks/p1/p1_provisioning_test.go

```diff
@@ -206,6 +226,54 @@ var _ = Describe("P1Provisioning", func() {
 				updateClusterInUpdatingState(cluster, ctx.RancherAdminClient, upgradeToVersion)
 			})
 		})
+
+		When("a cluster is created with multiple nodegroups", func() {
+			BeforeEach(func() {
+				var err error
+				createFunc := func(clusterConfig *eks.ClusterConfig) {
+					*clusterConfig, err = helper.AddNodeGroupToConfig(*clusterConfig, 4)
+					Expect(err).To(BeNil())
+				}
+				cluster, err = helper.CreateEKSHostedCluster(ctx.RancherAdminClient, clusterName, ctx.CloudCredID, k8sVersion, region, createFunc)
+				Expect(err).To(BeNil())
+				cluster, err = helpers.WaitUntilClusterIsReady(cluster, ctx.RancherAdminClient)
+				Expect(err).To(BeNil())
+			})
+
+			It("Update k8s version of node groups - sequential & simultaneous upgrade of multiple node groups", func() {
+				testCaseID = 153
+				var err error
+				cluster, err = helper.UpgradeClusterKubernetesVersion(cluster, upgradeToVersion, ctx.RancherAdminClient, true)
+				Expect(err).To(BeNil())
+
+				// upgrade 2 nodegroups simultaneously
+				updateFunc := func(cluster *management.Cluster) {
+					nodeGroups := cluster.EKSConfig.NodeGroups
+					for i := 0; i <= 1; i++ {
+						nodeGroups[i].Version = &upgradeToVersion
+					}
+				}
+				cluster, err = helper.UpdateCluster(cluster, ctx.RancherAdminClient, updateFunc)
+				Expect(err).To(BeNil())
+				err = clusters.WaitClusterToBeUpgraded(ctx.RancherAdminClient, cluster.ID)
+				Expect(err).To(BeNil())
+				Eventually(func() bool {
+					cluster, err = helper.UpdateCluster(cluster, ctx.RancherAdminClient, updateFunc)
```

Comment:
Is `UpdateCluster` required again here ?

Reply:
Hmm, that shouldn't have been here. It is supposed to simply fetch the cluster. Let me fix that.

Reply:
https://github.com/rancher/hosted-providers-e2e/actions/runs/12927910879